components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsAuthenticatedUser.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js;
+
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+
+/**
+ * Javascript wrapper for Java level AuthenticatedUser.
+ * This provides controlled access to AuthenticatedUser object via provided javascript native syntax.
+ * e.g
+ * var userName = context.lastAuthenticatedUser.username
+ * <p>
+ * instead of
+ * var userName = context.getLastAuthenticatedUser().getUserName()
+ * <p>
+ * Also it prevents writing an arbitrary values to the respective fields, keeping consistency on runtime
+ * AuthenticatedUser.
+ *
+ * @see AuthenticatedUser
+ */
+public class JsAuthenticatedUser extends AbstractJSObjectWrapper<AuthenticatedUser> {
+
+    public JsAuthenticatedUser(AuthenticatedUser wrapped) {
+        this.wrapped = wrapped;
+    }
+
+    @Override
+    public Object getMember(String name) {
+        if (wrapped == null) {
+            return false;
```

Comment:
here if the wrapped AuthenticatedUser is null, we should return null object. We can use return super.getMember(name)